### PR TITLE
Update package for  octopus 14.1

### PR DESCRIPTION
**Pull request by iamashwin99**
- add new version hash
- pass `--enable-silent-deprecation` to configure to allow use of autotools

### CODE REVIEW

**Comment by iamashwin99** [comment]
berkeleygw 4 is released but cant be downloaded as their download links are unreachable.
Things to consider:
- does octopus compile with berkeleygw 4 at all?
- should we limit the ci to use berkeleygw3 which is already in spack source cache.

**Comment by iamashwin99** [comment]
explicitly setting berkeleygw@3.0.1 seems to work, so lets keep that for now

File: Makefile [hunk 1]
```diff
@@ -2,7 +2,9 @@
 
 # Using most recent spack version with preferred version of octopus
 spack-develop:
-	docker build -f Dockerfile --build-arg SPACK_VERSION=develop -t octopus-spack  .
+	# Setting BERKELEYGW_VER to 3.0.1 as v4 is released but the download link is broken!
```
Comment (fangohr) [inline Makefile:5]: I thought Octopus would prefer not to use v4 (but rather 3.10 - not sure about the precise number). Can you pdate the comment and version accordingly?

**Review by fangohr — APPROVED** [review]
Open open question, otherwise all good. Thanks!

File: Makefile (hunk 1 above)
Comment (iamashwin99) [reply Makefile:5]: Actually, version 4 was released just this month and Octopus developers themselves haven't tested this https://gitlab.com/octopus-code/octopus/-/issues/1003 I am reluctant to make changes to the `package.py` or make a comment here until the octopus developers themself confirm the support or absence of it. You are right that the previous recommended version here is v3.0.1 as is written in the comment.

**Pull request merged**